src/test/java/jenkins/bouncycastle/SecurityProviderInitializerTest.java

```diff
@@ -25,20 +25,32 @@
 package jenkins.bouncycastle;
 
 import static org.junit.Assert.assertNotNull;
-
+import static org.junit.Assert.assertNull;
 import java.security.Security;
-
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.jvnet.hudson.test.JenkinsRule;
 
 public class SecurityProviderInitializerTest {
 
+    @BeforeClass
+    public static void validateEnv() {
+        assertNull(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME));
```

Comment:
this was not checking that BC was not alread registered, and so was not actually checking the code was registering BouncyCastle :)